### PR TITLE
Issue-9402: Added ImmediateTriggerer class

### LINKED ISSUE #9402: Create a trigger immediately and once batch schedule trigger
**Is your enhancement request related to a problem? Please describe.**
Extend the selection of batch scheduler triggers to include a trigger immediately and once batch schedule trigger 

**Describe the solution you'd like**
A batch function scheduler that will schedule the function to run immediately upon submission

**Describe alternatives you've considered**
Using a CRON scheduler, but it periodically re-schedules the function to run again

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by david-streamlio**
Fixes #9402 

We needed a new type of tirggerer for batch sources that is triggered exactly once, e.g. a S3 source that receives CDC notifications via an SQS topic. In this scenario, the discover phase only needs to run once to setup the SQS connection, etc.

<!--
### Contribution Checklist

*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #9402 

### Motivation
We needed a new type of tirggerer for batch sources that is triggered exactly once, e.g. a S3 source that receives CDC notifications via an SQS topic. In this scenario, the discover phase only needs to run once to setup the SQS connection, etc.

### Modifications

Added a new class called ImmediateTriggerer

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:
NO

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no) YES
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented). JavaDocs
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by jiazhai** [comment]
@david-streamlio would you please help provide some description for this pr?

**Review by jerrypeng — APPROVED** [review]

**Review by sijie — CHANGES_REQUESTED** [review]
@david-streamlio how this trigger will be used? Any documentation or test cases?

**Comment by david-streamlio** [comment]
@sijie An example use case would be an S3 source connector that uses SQS to get event notifications of changes rather than scanning the entire S3 bucket for changes.  https://docs.aws.amazon.com/AmazonS3/latest/dev/NotificationHowTo.html

In such a scenario, the discover phase of the source would connect to the SQS queue as described [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/ways-to-add-notification-config-to-bucket.html), and only need to execute once.

**Comment by sijie** [comment]
@david-streamlio is it possible to add an integration test for this trigger?

**Comment by david-streamlio** [comment]
@sijie I don't see any unit test or integration tests for the existing `CronTriggerer` class so I am unsure how to go about testing this class.

File: pulsar-broker/src/test/java/org/apache/pulsar/io/PulsarFunctionE2ETest.java [hunk 1]
```diff
@@ -266,6 +270,28 @@ void setup(Method method) throws Exception {
...
+                fileServer.createContext("/pulsar-io-batch-data-generator.nar", he -> {
+                    try {
+
+                        Headers headers = he.getResponseHeaders();
+                        headers.add("Content-Type", "application/octet-stream");
+
+                        File file = new File(getClass().getClassLoader().getResource("pulsar-io-batch-data-generator.nar").getFile());
+                        byte[] bytes  = new byte [(int)file.length()];
+
+                        FileInputStream fileInputStream = new FileInputStream(file);
```
Comment (eolivelli) [inline pulsar-broker/src/test/java/org/apache/pulsar/io/PulsarFunctionE2ETest.java:282]: please close this resource

**Review by eolivelli — COMMENTED** [review]
can you please rebase this PR onto current master ?
The diff shows many commits that are not from this patch

File: pulsar-broker/src/test/java/org/apache/pulsar/io/PulsarFunctionE2ETest.java (hunk 1 above)
Comment (david-streamlio) [reply pulsar-broker/src/test/java/org/apache/pulsar/io/PulsarFunctionE2ETest.java:282]: i have rebased onto the current master branch and added code to close all the opened input streams in the code

**Comment by david-streamlio** [comment]
Replacing the https://github.com/apache/pulsar/pull/9727

**Pull request closed without merging**